src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py

```diff
@@ -163,6 +164,11 @@ def _get_aad_token_after_challenge(cli_ctx,
     response = requests.post(authhost, urlencode(content), headers=headers,
                              verify=(not should_disable_connection_verify()))
 
+    if response.status_code == 429:
+        if is_diagnostics_context:
+            return CONNECTIVITY_TOOMANYREQUESTS_ERROR.format_error_message(login_server)
+        raise CLIError(CONNECTIVITY_TOOMANYREQUESTS_ERROR.format_error_message(login_server)
+                       .get_error_message())
```

Comment:
Could you please use the specific error type from `azure.cli.core.azclierror` instead of `CLIError`